baybe/surrogates/bandit.py

```diff
@@ -0,0 +1,153 @@
+            rates, where ``N`` is the number of bandit arms.
+            Contains ``float('nan')`` for arms with undefined mode.
+        """
+        from torch.distributions import Beta
+
+        return Beta(*self._posterior_beta_parameters()).mode
+
+    def posterior_means(self) -> Tensor:
+        """Compute the posterior mean win rates for all arms.
+
+        Returns:
+            A tensor of length ``N`` containing the posterior mean estimates of the win
+            rates, where ``N`` is the number of bandit arms.
+        """
+        from torch.distributions import Beta
+
+        return Beta(*self._posterior_beta_parameters()).mean
+
+    def _posterior_beta_parameters(self) -> Tensor:
+        """Compute the posterior parameters of the beta distribution.
+
+        Raises:
+            ModelNotTrainedError: If accessed before the model was trained.
+
+        Returns:
+            A tensors of shape ``(2, N)`` containing the posterior beta parameters,
+            where ``N`` is the number of bandit arms.
+        """
+        if self._win_lose_counts is None:
+            raise ModelNotTrainedError(
+                f"'{self.__class__.__name__}' must be trained before posterior "
+                f"information can be accessed."
+            )
+        return self._win_lose_counts + self.prior.to_tensor().unsqueeze(-1)
+
+    def to_botorch(self) -> Model:  # noqa: D102
+        # See base class.
+
+        # We register the sampler on the fly to avoid eager loading of torch
+
+        from botorch.sampling.base import MCSampler
+        from botorch.sampling.get_sampler import GetSampler
+        from torch.distributions import Beta
+
+        class CustomMCSampler(MCSampler):
+            """Customer sampler for beta posterior."""
+
+            def forward(self, posterior: TorchPosterior) -> Tensor:
+                """Sample the posterior."""
+                with temporary_seed(self.seed):
+                    samples = posterior.rsample(self.sample_shape)
+                return samples
+
+        @GetSampler.register(Beta)
+        def get_custom_sampler(_, sample_shape, seed: int | None = None):
+            """Get the sampler for the beta posterior."""
+            return CustomMCSampler(sample_shape=sample_shape, seed=seed)
+
+        return super().to_botorch()
+
+    @staticmethod
+    def _make_target_scaler_factory():
+        # See base class.
+        #
+        # We directly use the binary computational representation from the target.
+        return None
+
+    def _posterior(self, candidates: Tensor, /) -> TorchPosterior:
+        # See base class.
+
+        from botorch.posteriors import TorchPosterior
+        from torch.distributions import Beta
+
+        beta_params_for_candidates = self._posterior_beta_parameters().T[
+            candidates.argmax(-1)
+        ]
+        return TorchPosterior(Beta(*beta_params_for_candidates.split(1, -1)))
```

Comment:
@julianStreibel: I'm a a bit confused about the necessary last dimension of the tensor here. Perhaps we can talk about it in a meeting